src/main/java/com/b44t/messenger/rpc/EnteredLoginParam.java

```diff
@@ -0,0 +1,171 @@
+
+    // If true, login via OAUTH2 (not recommended anymore)
+    private final boolean oauth2;
+
+    public EnteredLoginParam(String addr,
+                             String password,
+                             String imapServer,
+                             int imapPort,
+                             SocketSecurity imapSecurity,
+                             String imapUser,
+                             String smtpServer,
+                             int smtpPort,
+                             SocketSecurity smtpSecurity,
+                             String smtpUser,
+                             String smtpPassword,
+                             EnteredCertificateChecks certificateChecks,
+                             boolean oauth2) {
+        this.addr = addr;
+        this.password = password;
+        this.imapServer = imapServer;
+        this.imapPort = imapPort;
+        this.imapSecurity = imapSecurity;
+        this.imapUser = imapUser;
+        this.smtpServer = smtpServer;
+        this.smtpPort = smtpPort;
+        this.smtpSecurity = smtpSecurity;
+        this.smtpUser = smtpUser;
+        this.smtpPassword = smtpPassword;
+        this.certificateChecks = certificateChecks;
+        this.oauth2 = oauth2;
+    }
+
+    public String getAddr() {
+        return addr;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getImapServer() {
+        return imapServer;
+    }
+
+    public int getImapPort() {
+        return imapPort;
+    }
+
+    public SocketSecurity getImapSecurity() {
+        return imapSecurity;
+    }
+
+    public String getImapUser() {
+        return imapUser;
+    }
+
+    public String getSmtpServer() {
+        return smtpServer;
+    }
+
+    public int getSmtpPort() {
+        return smtpPort;
+    }
+
+    public SocketSecurity getSmtpSecurity() {
+        return smtpSecurity;
+    }
+
+    public String getSmtpUser() {
+        return smtpUser;
+    }
+
+    public String getSmtpPassword() {
+        return smtpPassword;
+    }
+
+    public EnteredCertificateChecks getCertificateChecks() {
+        return certificateChecks;
+    }
+
+    public boolean isOauth2() {
+        return oauth2;
+    }
+
+    public enum EnteredCertificateChecks {
+        automatic, strict, acceptInvalidCertificates,
+    }
+
+    public static EnteredCertificateChecks certificateChecksFromInt(int position) {
+        switch (position) {
+            case 0:
+                return EnteredCertificateChecks.automatic;
+            case 1:
+                return EnteredCertificateChecks.strict;
+            case 2:
+                return EnteredCertificateChecks.acceptInvalidCertificates;
+        }
+        throw new IllegalArgumentException("Invalid certificate position: " + position);
+    }
+
+    public enum SocketSecurity {
+        // Unspecified socket security, select automatically.
+        automatic,
+
+        // TLS connection.
+        ssl,
```

Comment:
is this properly converted to string? sounds to me that the enum values would be converted to int while the socket security config expects a string, right? :thinking: